library/ccm.c

```diff
@@ -183,7 +207,11 @@ static int ccm_auth_crypt( mbedtls_ccm_context *ctx, int mode, size_t length,
     volatile unsigned char *output_dup = output;
     volatile unsigned char *tag_dup = tag;
     volatile size_t tag_len_dup = tag_len;
-
+    volatile unsigned int flow_ctrl_local = 0;
+    volatile int mode_dup = mode;
+    volatile int fc0 = FLAG_ERROR, fc1 = FLAG_ERROR, fc2 = FLAG_ERROR,
+                 fc3 = FLAG_ERROR, fc4 = FLAG_ERROR, fc5 = FLAG_ERROR,
+                 fc6 = FLAG_ERROR, fc7 = FLAG_ERROR;
```

Comment:
Wouldn't it be better to make one extra flow counter than several `int` variables? Such a counter would be incremented by one, sometimes by 2 and sometimes by 3

Reply:
The thing is that the amount of control flow incrementations depends on function flow. The best way to know at the end of the function what the value of the flow_control should be - would be... going through the function flow with the same data.

Reply:
@AndrzejKurek , Please add comments to the function that explain this FI return codes and the reason for chosen implementation, otherwise we will explain it again and again

Reply:
Done.